Sources/ProjectDescription/Settings.swift

```diff
@@ -124,12 +124,60 @@ public extension CustomConfiguration {
 /// The default settings can be overridden via `Settings base: SettingsDictionary`
 /// and `Configuration settings: SettingsDictionary`.
 ///
-/// - all: Essential settings plus all the recommended settings (including extra warnings)
+/// - `recommended`: Essential settings plus all the recommended settings (including extra warnings)
 /// - essential: Only essential settings to make the projects compile (i.e. `TARGETED_DEVICE_FAMILY`)
-public enum DefaultSettings: String, Codable {
+indirect public enum DefaultSettings: Codable, Equatable {
     case recommended
     case essential
     case none
+    case excluding(DefaultSettings, Set<String>)
+
+    enum CodingKeys: CodingKey {
+        case recommended, essential, none, excluding
+    }
+
+    public init(from decoder: Decoder) throws {
```

Comment:
Seeing there's some custom encode / decode logic, worth updating

https://github.com/tuist/tuist/blob/master/Tests/ProjectDescriptionTests/SettingsTests.swift#L7

to include a case that leverages the `excluded` keys

Reply:
added test